Lesson3

Northman94 opened the pull request.

Northman94 committed:
Replaced venvL2 to venv in the Proj Root

Northman94 committed:
L3 main creation

Northman94 committed:
Added form items and receiving func blueprint

Northman94 committed:
Created main logic with html blueprint

Northman94 committed:
Added house template & 3 endpoints

Northman94 committed:
Added Page skipping check to avoid cheating

Northman94 committed:
SECRET_KEY changed and stored locally

+ fixed Check on page skipping
+ added ascii() check in constructor
+ secret key changed and stored separatelly

idegtiarov commented on Lesson3/main_L3.py:
```diff
@@ -0,0 +1,108 @@
+
+class User():
+    def __init__(self, name='', language='', house='', magic_item_level=''):
+        self.name = ascii(name)
```
`ascii` function provides no validation but escapes all non ascii symbols, please take a look at https://docs.python.org/3/library/functions.html#ascii.

idegtiarov commented on Lesson3/main_L3.py:
```diff
@@ -0,0 +1,108 @@
+class User():
+    def __init__(self, name='', language='', house='', magic_item_level=''):
+        self.name = ascii(name)
+        self.language = ascii(language)
```
ditto

idegtiarov commented on Lesson3/main_L3.py:
```diff
@@ -0,0 +1,108 @@
+
+@app.route("/magic_item", methods=["GET", "POST"])
+def magic_item():
+    name = request.args.get('name')
```
We could omit to check query parameter all that data is already should be added to the User instance as valid attributes

idegtiarov reviewed (changes requested):
Please take a look at my comments, other LGTM!

Northman94 committed:
Ascii symbol check

Northman94 committed:
added check_input func

Northman94 committed:
attempt to store variables in class.

not finished

Northman94 committed:
User info is now stored in single User. Not passed from first Route to Second to Third.
In future should replace init with just variables, because it is initializing before first data being submitted.
